File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

         if (s.length() != t.length()) {
            return false;
        }

        char[] sArray = s.toCharArray();
        char[] tArray = t.toCharArray();
        Arrays.sort(sArray);
        Arrays.sort(tArray);
        return Arrays.equals(sArray, tArray);
    }

    public static void main(String[] args) {
        Solution va = new Solution();
        System.out.println(va.isAnagram("anagram", "nagaram")); 
        System.out.println(va.isAnagram("rat", "car"));         
        
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()){
            return false;
        }
        unordered_map<char,int>m;

        for (char c : s){
            m[c]++;

        }
        for (auto c2 : t){
            if (m.find(c2)==m.end()){
                return false;
            }
            else {
                m[c2]--;
            }
        }
        for(auto ele:m){
            if (ele.second != 0){
                return false;
            }
            
        }return true;
        
    }
};